scripts/deploy/cluster-up.sh

```diff
@@ -0,0 +1,300 @@
+else
+  AZURE_CLUSTER_TEMPLATE_FILE=${AZURE_CLUSTER_TEMPLATE}
+fi
+
+if [[ -z ${AZURE_CLUSTER_DNS_NAME:-} ]]; then
+  CLUSTER_PREFIX=$(whoami)
+  if [[ ${CLUSTER_PREFIX:-root} == "root" ]]; then
+    CLUSTER_PREFIX=k8s
+  fi
+  AZURE_CLUSTER_DNS_NAME=$(basename "$(mktemp -t "${CLUSTER_PREFIX}-${AZURE_CLUSTER_TEMPLATE}-${GIT_COMMIT}-XXX")")
+fi
+
+if [[ -z ${AZURE_RESOURCE_GROUP:-} ]]; then
+  AZURE_RESOURCE_GROUP=${AZURE_CLUSTER_DNS_NAME}-rg
+fi
+
+if [[ -z ${OUTPUT_DIR:-} ]]; then
+  OUTPUT_DIR="$GIT_ROOT/_output/$AZURE_CLUSTER_DNS_NAME"
+fi
+
+#
+# Install required tools
+#
+install_helm
+
+#
+# Create the Kubernetes cluster
+#
+echo "Creating cluster ${AZURE_CLUSTER_DNS_NAME}"
+"${GIT_ROOT}/scripts/deploy/azure-cluster-up.sh" \
+  --subscription "${AZURE_SUBSCRIPTION_ID}" \
+  --location "${AZURE_LOCATION}" \
+  --name "${AZURE_CLUSTER_DNS_NAME}" \
+  --output "${OUTPUT_DIR}" \
+  --k8s-version "${AZURE_K8S_VERSION}" \
+  --template "${AZURE_CLUSTER_TEMPLATE}"
+
+# Delete the cluster on subsequent errors.
+trap_push "\"${OUTPUT_DIR}/cluster-down.sh\"" err
+
+echo "Wait for cluster to become available..."
+# TODO Figure out a better way to determine cluster availability
+sleep 5m
+
+export KUBECONFIG="${OUTPUT_DIR}/kubeconfig/kubeconfig.${AZURE_LOCATION}.json"
+
+#
+# Install the Azure Disk CSI Driver
+#
+# echo "Installing Azure Disk CSI Driver..."
+# helm repo add azuredisk-csi-driver https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/master/charts
+# helm upgrade \
+#   --install \
+#   --namespace kube-system \
+#   azuredisk-csi-driver \
+#   azuredisk-csi-driver/azuredisk-csi-driver \
+#   -f "${AZURE_CLUSTER_TEMPLATE_ROOT}/${AZURE_CLUSTER_TEMPLATE}/azuredisk-csi-driver-values.yaml"
+# echo "Waiting for Azure Disk CSI Driver to start..."
+# kubectl wait pods --namespace kube-system --selector app.kubernetes.io/instance=azuredisk-csi-driver --for condition=ready --timeout=15m
+
+#
+# Apply label to Mayastor Node Candidates
+#
+echo "Labelling Mayastor Node Candidates..."
+kubectl label nodes --selector agentpool=agentpool openebs.io/engine=mayastor 
+
+#
+# Enable hugepages and prepare the nodes for reboot
+#
+echo "Enabling HugePages and marking all the nodes for reboot..."
+kubectl apply -f actions/hugepage-enabler-daemonset.yaml
+sleep 2m
+
+#
+# Restart nodes
+#
+echo "Restarting nodes..."
+kubectl apply -f actions/kured-config.yaml
+sleep 3m
+getNodesStatus
+echo "Nodes restart successful"
+
+#
+# Creating Mayastor Application Namespace and Resources
+#
+echo "Starting Mayastor specific operations..."
+kubectl create namespace mayastor
+kubectl apply -f https://raw.githubusercontent.com/openebs/mayastor-control-plane/master/deploy/operator-rbac.yaml
+kubectl apply -f https://raw.githubusercontent.com/openebs/mayastor-control-plane/master/deploy/mayastorpoolcrd.yaml
+
+#
+# Deploy NATS
+#
+kubectl apply -f https://raw.githubusercontent.com/openebs/mayastor/master/deploy/nats-deployment.yaml
+sleep 2m
```

Comment:
Consider the following command instead of the sleep and watch pod status:

```bash
kubectl rollout status statefulset/nats --namespace mayastor --watch
```